plugins/functions/string.c

```
@@ -294,6 +304,165 @@ func_string_tokenize(grn_ctx *ctx, int n_args, grn_obj **args,
+                         GRN_TEXT_VALUE(pattern),
+                         GRN_TEXT_LEN(pattern),
+                         GRN_ONIGMO_OPTION_DEFAULT,
+                         GRN_ONIGMO_SYNTAX_DEFAULT,
+                         "[regex_slice][regexp]");
+
+  if (!regex) {
+    return NULL;
+  }
+
+  region = onig_region_new();
+
+  target_string = GRN_TEXT_VALUE(target);
+  target_length = GRN_TEXT_LEN(target);
+
+  //Cannot use normalized string. 
+  //The matching parts of original string cannot be inferred from the matching parts of the normalized string.
+  position = onig_search(regex,
+                         target_string,
+                         target_string + target_length,
+                         target_string,
+                         target_string + target_length,
+                         region,
+                         0);
+
+  if (grn_obj_is_text_family_bulk(ctx, capture)) {
+    capture_name = GRN_TEXT_VALUE(capture);
+    capture_name_length = GRN_TEXT_LEN(capture);
+    capture_num = onig_name_to_backref_number(regex,
+                                              capture_name,
+                                              capture_name + capture_name_length,
+                                              region);
+  } else if (grn_obj_is_number_family_bulk(ctx, capture)) {
+    capture_num = grn_plugin_proc_get_value_int64(ctx,
+                                                  capture,
+                                                  0,
+                                                  "[regex_slice][capture_num]");
+  }
+
+  if (position >= 0 && capture_num >= 0 && region->num_regs > capture_num) {
+    start = region->beg[capture_num];
+    end = region->end[capture_num];
+    result = grn_plugin_proc_alloc(ctx, user_data, target->header.domain, 0);
+    if (result) {
+      GRN_BULK_REWIND(result);
+      GRN_TEXT_SET(ctx, result, target_string + start, end - start);
+    } else {
+      result = default_value;
```

Should we report an error for this case?